core/src/main/scala/io/finch/request/package.scala

```diff
@@ -309,7 +358,9 @@ package object request extends LowPriorityImplicits {
    *
    * @param message the message
    */
-  class RequestReaderError(val message: String) extends Exception(message)
+  class RequestReaderError(val message: String, cause: Throwable) extends Exception(message, cause) {
+    def this(message: String) = this(message, null)
```

Comment:
I did not like this either, but this ugliness is caused by interfacing with Java. The `null` is only used when calling a Java constructor, and the base class is not a case class, so it is not accessible from Scala code. And in Java Exception getCause can return `null`: http://docs.oracle.com/javase/7/docs/api/java/lang/Throwable.html#getCause()

So as ugly as it is, I'm afraid it is the right thing to do in this case.